pyart/graph/radarmapdisplay_cartopy.py

```diff
@@ -0,0 +1,510 @@
+"""
+pyart.graph.radarmapdisplay_cartopy
+===================================
+
+Class for creating plots on a geographic map using a Radar object using Cartopy
+for drawing maps.
+
+.. autosummary::
+    :toctree: generated/
+    :template: dev_template.rst
+
+    RadarMapDisplayCartopy
+
+"""
+
+import numpy as np
+try:
+    import cartopy
+    _CARTOPY_AVAILABLE = True
+except ImportError:
+    _CARTOPY_AVAILABLE = False
+
+from .radardisplay import RadarDisplay
+from .common import parse_ax_fig, parse_norm_vmin_vmax, parse_cmap
+from ..exceptions import MissingOptionalDependency
+
+
+class RadarMapDisplayCartopy(RadarDisplay):
+    """
+    A display object for creating plots on a geographic map from data in a
+    Radar object.
+
+    This class is still a work in progress.  Some functionality may not work
+    correctly. Please report any problems to the Py-ART GitHub Issue Tracker.
+
+    Parameters
+    ----------
+    radar : Radar
+        Radar object to use for creating plots.
+    shift : (float, float)
+        Shifts in km to offset the calculated x and y locations.
+
+    Attributes
+    ----------
+    plots : list
+        List of plots created.
+    plot_vars : list
+        List of fields plotted, order matches plot list.
+    cbs : list
+        List of colorbars created.
+    origin : str
+        'Origin' or 'Radar'.
+    shift : (float, float)
+        Shift in meters.
+    loc : (float, float)
+        Latitude and Longitude of radar in degrees.
+    fields : dict
+        Radar fields.
+    scan_type : str
+        Scan type.
+    ranges : array
+        Gate ranges in meters.
+    azimuths : array
+        Azimuth angle in degrees.
+    elevations : array
+        Elevations in degrees.
+    fixed_angle : array
+        Scan angle in degrees.
+    proj : Proj
+        Object for performing cartographic transformations specific to the
+        geographic map plotted.
+    basemap : Basemap
+        Last plotted basemap, None when no basemap has been plotted.
+
+
+    """
+
+    def __init__(self, radar, shift=(0.0, 0.0)):
+        """ Initialize the object. """
+        # check that basemap is available
+        if not _CARTOPY_AVAILABLE:
+            raise MissingOptionalDependency(
+                "Cartopy is required to use RadarMapDisplayCartopy but is " +
+                "not installed")
+
+        # initalize the base class
+        RadarDisplay.__init__(self, radar, shift=shift)
+
+        # additional attributes needed for plotting on a basemap.
+        self.basemap = None
```

Comment:
Can we change the name of this attribute.  It does not refer to a Basemap instance so the name is poor.  Perhaps crs since this will be a Cartopy coordinate reference system?